src/ClientInstance.cpp

```diff
@@ -384,14 +385,22 @@ PVR_ERROR CVNSIClientInstance::GetChannels(bool radio, kodi::addon::PVRChannelsR
       if (m_protocol >= 6)
       {
         std::string path = CVNSISettings::Get().GetIconPath();
-        std::string ref = vresp->extract_String();
+        std::string ref = vresp->extract_String(); // Picon ref
         if (!path.empty())
         {
           if (path[path.length() - 1] != '/')
             path += '/';
-          path += ref;
-          path += ".png";
-          tag.SetIconPath(path);
+
+          ref += ".png";
+          std::string image_path = path + ref;
+          if (!kodi::vfs::FileExists(image_path)) // Use channel name
+          {
+            std::string channelNameImage = tag.GetChannelName();
+            std::transform(channelNameImage.begin(), channelNameImage.end(), channelNameImage.begin(), ::tolower);
+            channelNameImage += ".png";
+            std::string image_path = path + channelNameImage;
```

Comment:
drop `std::string ` here. The variable already exists and this will just crafted a new variable that will be destroyed and your code won’t work.

Reply:
Maybe now ok? I'm little bit newbie with c++ 😄

Reply:
Yes, it looks good now.